src/gamelogic.cpp

```diff
@@ -150,7 +150,7 @@ WinnerID playGame(BoardType& playerBoard, BoardType& computerBoard) {
                 return potentialWinnerID;
         }
 
-        if (!playerHitOnce) {
+        if (!playerHit) {
```

Comment:
Revert the changes to the while loop and simply get rid of the if condition for the code under it